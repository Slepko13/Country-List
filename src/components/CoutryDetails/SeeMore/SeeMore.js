import React, { useState} from 'react';
import './SeeMore.scss';



const SeeMore = ({data,top, bottom,left,right})=>{
    const [list, setList] = useState(false);
    console.log(list);
return (
    <div
        className="SeeMore"
        style={{bottom,top,right,left}}
    >
        {!list ? <button className="openButton" onClick={()=>setList(true)}>See more</button>:
            data.map(item=>
                <div
                    className="item__see"
                    key={item.name}
                >{item.name.replace("UTC","GMT")}</div>
            )
        }
        {list && <button className="closeButton" onClick={()=>setList(false)}>Close</button>}
    </div>
)
}

export default SeeMore;