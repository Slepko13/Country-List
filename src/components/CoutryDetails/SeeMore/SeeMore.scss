.SeeMore{
  position: absolute;
  left :0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  z-index: 3;
  background-color: #ffffff;

  .item__see {
    font-family: "Roboto";
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 20px;
    color: #313131;
  }
  .openButton,
  .closeButton {
    font-family: "Roboto";
    font-style: normal;
    font-weight: bold;
    font-size: 12px;
    line-height: 14px;
    margin-top: 5px;
    border-radius: 5px;
    border: none;
    background: linear-gradient(180deg, rgba(255, 228, 133, 0.5) 21.88%, rgba(186, 123, 0, 0.5) 100%);
    cursor: pointer;
  }

}