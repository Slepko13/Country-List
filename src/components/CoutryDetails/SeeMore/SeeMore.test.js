import React from "react";
import SeeMore from "./SeeMore";

const setUp = (props) => shallow(<SeeMore{...props}/>)
const props = {
    data : [
        {name :"english" },
        {name :"french" },
    ],
    left:"0",
    right:"0",
    top:"0",
    bottom:"0"
}

describe("my SeeMore component", () => {
    it("should render SeeMore  Component with props(snapshot)", () => {
        const component = setUp(props);
        expect(component).toMatchSnapshot();
    });
    it("should  show close  button", () => {
        const setList = jest.fn();
        const component = setUp(props);
        component.find(".openButton").simulate('click');
        expect(component.find("closeButton")).not.toBeNull();
    });
    it("should hide close  button", () => {
        const setList = jest.fn();
        const component = setUp(props);
        component.find(".openButton").simulate('click');
        component.find(".closeButton").simulate('click');
        expect(component.find("openButton")).not.toBeNull();
        expect(component.find("closeButton")).toHaveLength(0);

    });
});
