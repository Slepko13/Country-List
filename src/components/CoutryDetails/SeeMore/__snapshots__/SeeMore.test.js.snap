// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my SeeMore component should render SeeMore  Component with props(snapshot) 1`] = `
<div
  className="SeeMore"
  style={
    Object {
      "bottom": "0",
      "left": "0",
      "right": "0",
      "top": "0",
    }
  }
>
  <button
    className="openButton"
    onClick={[Function]}
  >
    See more
  </button>
</div>
`;

exports[`my SeeMore component should render SeeMore  list 1`] = `
<div
  className="SeeMore"
  style={
    Object {
      "bottom": "0",
      "left": "0",
      "right": "0",
      "top": "0",
    }
  }
>
  <button
    className="openButton"
    onClick={[Function]}
  >
    See more
  </button>
</div>
`;
